static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -415,6 +415,73 @@ class SpanTreeModel {
 
           // This may not be the case, and needs to be looked into later
 
+          const key = getSiblingGroupKey(group[0].span);
+          if (this.expandedSiblingGroups.has(key)) {
+            // Check if any of the spans in the sibling group are filtered out (possible when filtering by span ID)
+            group.forEach((spanModel, index) => {
+              if (this.isSpanFilteredOut(props, spanModel)) {
```

Comment:
Maybe expand on why/how the parent can be unfiltered but the children can be filtered? Or is this an issue of `return` ordering? If you move the filter condition below above this one, does it affect how it's display? If we're filtering out the parent do we still show children?